src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+        return record
+
+    @staticmethod
+    def _get_score(verdict_output: str, statements: str) -> float:
+        """Given generated statements and verdicts, compute Faithfulness score.
+
+        :param verdict_output: Verdicts(Yes/No) and explanations string get from Judge model.
+        :param statements: Statements string get from `GetStatements` Transform.
+        :returns: 0 to 1. See the docstring for `Faithfulness` for more details
+            on what these numerical values represent.
+        """
+        output = verdict_output.lower().strip()
+        num_statements = len(statements.split("\n"))
+        # TODO: handle edge case that num_statements is 0?
+
+        # calculate score
+        final_answer = "Final verdicts in order:"
+        final_answer = final_answer.lower()
+        if output.find(final_answer) != -1:
+            output = output[output.find(final_answer) + len(final_answer) :]
+            score = float(sum(1 if "yes" in answer else 0 for answer in output.strip().split(".") if answer != ""))
+            score = score / num_statements
+        else:
+            score = float(max(0, output.count("verdict: yes")) / num_statements)
+
+        return score
+
+
+class GetStatements(Transform):
+    """This transform invokes the judge model string from raw statements."""
+
+    def __init__(self, input_key: str, output_key: str, judge_model: ModelRunner):
+        """GetStatements Initializer.
+        :param input_key: The key corresponding to prompt to get statements.
+        :param output_key: The key corresponding to the statements, which gets added to the record.
+        :param judge_model: An instance of ModelRunner representing the judge model to be used.
+            If this argument is None, default judge model will be provided.
+        """
+        super().__init__(input_key, output_key, judge_model)
+        self.register_input_output_keys(
+            input_keys=[input_key],
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and arguments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the mean added in.
+        """
+        get_raw_statements = GetModelOutputs(
+            input_to_output_keys={self.input_key: ["raw_statements"]},
+            model_runner=self.judge_model,
+        )
+        record_with_raw_statements = get_raw_statements(record)
+        raw_statements = record_with_raw_statements["raw_statements"].split("\n")
+        list_statements = [statement for statement in raw_statements if statement.startswith("Statement:")]
+        statements_str: str = "\n".join([f"{i + 1}.{st}" for i, st in enumerate(list_statements)])
```

Comment:
The issue with Ray not being able to support lists natively seems to be solved in the latest version of Ray: https://github.com/ray-project/ray/issues/39559#issuecomment-2138390318

Can you see if you can achieve the required results by storing these statements as a list of statements, instead of a string?